### PR TITLE
Remove Factory and LinuxFactory

### PR DESCRIPTION
**Pull request by kolyshkin**
~~_Based on and currently includes #3374. Draft until that one is merged._~~

Continuing the spring cleaning started in #3354. This dismantles Factory and LinuxFactory, doing some cleanups along the way.

Loosely based on parts of #3316.

### CODE REVIEW

**Comment by kolyshkin** [comment]
No longer a draft; PTAL @opencontainers/runc-maintainers @thaJeztah @AkihiroSuda

**Comment by kolyshkin** [comment]
PTAL @opencontainers/runc-maintainers @thaJeztah @AkihiroSuda (I have a few more draft PRs on top of this).

**Comment by kolyshkin** [comment]
This is mostly code removal, and is organized to ease the review as much as possible.

**Comment by kolyshkin** [comment]
Anything I can do to move this forward, @opencontainers/runc-maintainers? I have a few more PRs based on top of this.

**Review by crosbymichael — DISMISSED** [review]

File: libcontainer/factory_linux.go [hunk 1]
```diff
@@ -28,60 +27,35 @@ const (
...
+// used later to get the list of containers, or to get information about a
+// particular container (see Load).
+//
+// The id must not be empty and consists of only the following characters:
+// ASCII letters, digits, underscore, plus, minus, period. The id must be
+// unique and non-existent for the given root path.
+//
+// Returns the new container with a running process.
+//
+// On error, any partially created container parts are cleaned up (the
```
Comment (AkihiroSuda) [inline libcontainer/factory_linux.go:43]: I don't see any clean-up code

File: libcontainer/factory_linux.go [hunk 2]
```diff
@@ -28,60 +27,35 @@ const (
...
+// operation is atomic).
```
Comment (AkihiroSuda) [inline libcontainer/factory_linux.go:44]: "atomic" might not be the right word.
"atomic" means "atomic" as in `sync/atomic` to mee.

**Comment by AkihiroSuda** [comment]
:+1: but godoc might not be correct

File: libcontainer/factory_linux.go (hunk 2 above)
Comment (kolyshkin) [reply libcontainer/factory_linux.go:44]: It's worse than that -- currently, `Create` does not even creates a container (with a process running inside it, as the doc says) -- it merely creates a container directory and returns some structure that can be used to actually start a container.

I'll try to modify the docs accordingly. These were just moved from factory to not lose them, but seems like it was not a good idea.

**Review by AkihiroSuda — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**